src/main/webapp/app/entities/text-exercise/text-exercise-update.component.ts

```diff
@@ -1,51 +1,136 @@
+
+import { ExerciseCategory, ExerciseService } from 'app/entities/exercise';
+import { ExampleSubmissionService } from 'app/entities/example-submission/example-submission.service';
+import { KatexCommand } from 'app/markdown-editor/commands';
+import { EditorMode } from 'app/markdown-editor';
+import { MAX_SCORE_PATTERN } from 'app/app.constants';
+import { AssessmentType } from 'app/entities/assessment-type';
 
 @Component({
     selector: 'jhi-text-exercise-update',
     templateUrl: './text-exercise-update.component.html',
+    styleUrls: ['./text-exercise-update.scss'],
 })
 export class TextExerciseUpdateComponent implements OnInit {
+    EditorMode = EditorMode;
+    AssessmentType = AssessmentType;
+
     textExercise: TextExercise;
     isSaving: boolean;
+    maxScorePattern = MAX_SCORE_PATTERN;
+    exerciseCategories: ExerciseCategory[];
+    existingCategories: ExerciseCategory[];
+    notificationText: string | null;
+
+    courses: Course[];
+
+    domainCommandsProblemStatement = [new KatexCommand()];
+    domainCommandsSampleSolution = [new KatexCommand()];
+    domainCommandsGradingInstructions = [new KatexCommand()];
 
-    constructor(private textExerciseService: TextExerciseService, private activatedRoute: ActivatedRoute) {}
+    constructor(
+        private jhiAlertService: JhiAlertService,
+        private textExerciseService: TextExerciseService,
+        private exerciseService: ExerciseService,
+        private courseService: CourseService,
+        private eventManager: JhiEventManager,
+        private exampleSubmissionService: ExampleSubmissionService,
+        private activatedRoute: ActivatedRoute,
+        private router: Router,
+    ) {}
 
     ngOnInit() {
-        this.isSaving = false;
+        // This is used to scroll page to the top of the page, because the routing keeps the position for the
+        // new page from previous page.
+        window.scroll(0, 0);
+
         this.activatedRoute.data.subscribe(({ textExercise }) => {
             this.textExercise = textExercise;
+            this.exerciseCategories = this.exerciseService.convertExerciseCategoriesFromServer(this.textExercise);
+            this.courseService.findAllCategoriesOfCourse(this.textExercise.course!.id).subscribe(
+                (categoryRes: HttpResponse<string[]>) => {
+                    this.existingCategories = this.exerciseService.convertExerciseCategoriesAsStringFromServer(categoryRes.body!);
+                },
+                (categoryRes: HttpErrorResponse) => this.onError(categoryRes),
+            );
         });
+
+        this.isSaving = false;
+        this.notificationText = null;
+        this.courseService.query().subscribe(
+            (res: HttpResponse<Course[]>) => {
+                this.courses = res.body!;
+            },
+            (res: HttpErrorResponse) => this.onError(res),
+        );
     }
 
     previousState() {
-        window.history.back();
+        if (this.textExercise.course) {
```

Comment:
Have we already fixed the documentation for this component in our other PR? Otherwise it would be nice, if you could add some documentation too all methods here.

Reply:
You are right, we need to have better documentation. Added JSDoc to all the files I touched with my changes